src/components/layouts/collection-page-layout.tsx

```diff
@@ -1094,4 +1204,92 @@ const transformSanityTopics = (topics: any) => {
   return topics.items
 }
 
+const DialogButton = ({title, buttonStyles, buttonText, children}: any) => {
+  const [showDialog, setShowDialog] = React.useState(false)
+
+  const openDialog = () => {
+    setShowDialog(true)
+  }
+  const closeDialog = () => {
+    setShowDialog(false)
+  }
+
+  return (
+    <>
+      <button className={buttonStyles && buttonStyles} onClick={openDialog}>
+        <LockIcon className="h-4 w-4 mr-2" /> {buttonText}
+      </button>
+      <DialogOverlay
+        isOpen={showDialog}
+        onDismiss={closeDialog}
+        className="bg-black/50 backdrop-blur-sm flex justify-center items-center p-4 xs:px-6 xs:py-8 z-50"
+      >
+        <DialogContent
+          aria-label="sign in to bookmark"
+          className="bg-white dark:bg-gray-900 shadow-lg rounded-lg max-w-screen-sm border dark:border-gray-800 relative p-3 xs:p-6 w-112"
```

Comment:
suggestion: instead of a custom one-off tailwind class to achieve this we might consider just hard coding the width value as a style property unless this will be a common need across the entire app

related: https://github.com/eggheadio/egghead-next/pull/864/files#r708555190